storage/innobase/buf/buf0buf.cc

```diff
@@ -742,6 +742,221 @@ bool buf_page_is_corrupted(bool check_lsn, const byte *read_buf,
 
 #ifndef UNIV_INNOCHECKSUM
 
+/** Memory Pressure */
+
+#ifdef __linux__
+/*
+  based off https://www.kernel.org/doc/html/latest/accounting/psi.html#pressure-interface
+  and https://www.kernel.org/doc/html/latest/admin-guide/cgroup-v2.html#memory
+*/
+#include <poll.h>
+#include <fstream>
+
+class mem_pressure
```

Comment:
Please move the `/**` Doxygen comment right before `class mem_pressure`, and consider including the whole comment there, in case someone is actually generating documentation from the source code and reading it:

```c++
#ifdef __linux__
# include <poll.h>
# include <fstream>

/** Memory Pressure

based on …
*/
class mem_pressure
```